contracts/contracts/core/ProviderRegistry.sol

```diff
@@ -121,7 +121,18 @@ contract ProviderRegistry is
     ) external nonReentrant onlyPreConfirmationEngine whenNotPaused {
         uint256 residualAmt = (amt * residualBidPercentAfterDecay * PRECISION) / PERCENT;
         uint256 penaltyFee = (residualAmt * uint256(feePercent) * PRECISION) / PERCENT;
-        require(providerStakes[provider] >= residualAmt + penaltyFee, "Insufficient funds to slash");
+        // if the provider's stake is less than the residual amount + penalty fee, we need to adjust the residual amount and penalty fee
+        // this is to prevent underflow and ensure the contract doesn't revert
+        // We also emit
+        if (providerStakes[provider] < residualAmt + penaltyFee) {
+            emit InsufficientFundsToSlash(provider, providerStakes[provider], residualAmt, penaltyFee);
+            if (providerStakes[provider] < residualAmt) {
+                penaltyFee = 0;
+                residualAmt = providerStakes[provider];
+            } else {
+                penaltyFee = providerStakes[provider] - residualAmt;
+            }
```

Comment:
we could simplify logic and use single conditional statement:
```
if (providerStake < residualAmt) {
        residualAmt = providerStake;
}
penaltyFee = providerStake - residualAmt;
```

Reply:
Please, I hope via-ir does that for us XD

Reply:
My concern was an underflow here.

Reply:
how?

Reply:
if (providerStake < residualAmt) -> e.g. `3 < 5`, `rsdAmt = 3`, `pntFee = 3 - 3 = 0`

if (providerStake > residualAmt) -> e.g. `6 > 5`, `pntFee = 6 - 5 = 1`

Reply:
+1 on original comment, that logic could replace the entire nested if-statement unless I'm missing something

Reply:
Oh my bad, forgot that your mutating the variable before doing the subtraction.

Reply:
I still think setting penatlyFee to 0 here gives more clarity. But since two people are stating otherwise. I will change it.